xgcm/test/test_grid.py

```diff
@@ -701,29 +710,32 @@ def test_multi_axis_input(all_datasets, func, periodic, boundary):
         xr.testing.assert_allclose(serial, full)
 
 
-@pytest.mark.parametrize("func", ["interp", "max", "min", "diff", "cumsum"])
-@pytest.mark.parametrize("periodic", ["True", "False", ["X"], ["Y"], ["X", "Y"]])
+@pytest.mark.parametrize(
+    "func",
+    ["interp", "max", "min", "diff", pytest.param("cumsum", marks=pytest.mark.xfail)],
+)
 @pytest.mark.parametrize(
     "boundary",
     [
         "fill",
-        # "extrapolate", # do we not support extrapolation anymore?
+        pytest.param("extrapolate", marks=pytest.mark.xfail),
         "extend",
         {"X": "fill", "Y": "extend"},
         {"X": "extend", "Y": "fill"},
     ],
 )
-def test_dask_vs_eager(all_datasets, func, periodic, boundary):
+def test_dask_vs_eager(all_datasets, func, boundary):
     ds, coords, metrics = datasets_grid_metric("C")
     grid = Grid(ds, coords=coords)
-
-    eager_result = grid.diff(ds.tracer, "X")
+    grid_method = getattr(grid, func)
+    eager_result = grid_method(ds.tracer, "X", boundary=boundary)
 
     ds = ds.chunk({"xt": 1, "yt": 1, "time": 1, "zt": 1})
     grid = Grid(ds, coords=coords)
-    dask_result = grid.diff(ds.tracer, "X").compute()
+    grid_method = getattr(grid, func)
+    dask_result = grid_method(ds.tracer, "X", boundary=boundary).compute()
 
-    xr.testing.assert_identical(dask_result, eager_result)
+    xr.testing.assert_allclose(dask_result, eager_result)
```

Comment:
I realised this test was supposed to test many different method but it was only testing `diff` - now fixed.